### PR TITLE
Feature/booking app advanced filters

### PR DESCRIPTION
**Pull request by DDavidPrime**

### CODE REVIEW

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import 'dart:async';
 
+import 'package:booking_app/feature_hotel/search/models/capacity_filter_data.dart';
+import 'package:booking_app/feature_hotel/search/models/date_range_filter_data.dart';
 import 'package:favorites_advanced_base/core.dart';
 import 'package:favorites_advanced_base/models.dart';
 import 'package:flutter/material.dart';
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:7]: @DDavidPrime do we need to import the material package into the business layer?

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart [hunk 2]
```diff
@@ -30,20 +39,23 @@ abstract class HotelListEvents {
...
-  Stream<DateTimeRange?> get dateRangeFilter;
+  Stream<String> get hotelsFound;
 
   @RxBlocIgnoreState()
   Stream<String> get queryFilter;
 
-  /// Returns when the data refreshing has completed
   @RxBlocIgnoreState()
-  Future<void> get refreshDone;
+  Stream<DateRangeFilterData> get dateRangeFilterData;
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:51]: @DDavidPrime do we really need to use @RxBlocIgnoreState() here ?

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart [hunk 3]
```diff
@@ -30,20 +39,23 @@ abstract class HotelListEvents {
...
 
   @RxBlocIgnoreState()
-  Stream<String> get hotelsFound;
+  Stream<CapacityFilterData> get capacityFilterData;
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:54]: @DDavidPrime do we really need to use @RxBlocIgnoreState() here ?

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart [hunk 4]
```diff
@@ -82,13 +95,31 @@ class HotelListBloc extends $HotelListBloc {
...
-  Stream<PaginatedList<Hotel>> get searchedHotels => _hotels;
+  Stream<PaginatedList<Hotel>> get hotels => _hotels;
 
   @override
-  Stream<DateTimeRange?> get dateRangeFilter => _$filterByDateRangeEvent;
+  Stream<String> get queryFilter => _$filterByQueryEvent;
 
   @override
-  Stream<String> get queryFilter => _$filterByQueryEvent;
+  Stream<DateRangeFilterData> get dateRangeFilterData =>
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:104]: @DDavidPrime can we move the implementation details to the extension file ?

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart [hunk 5]
```diff
@@ -82,13 +95,31 @@ class HotelListBloc extends $HotelListBloc {
...
+        (range) => DateRangeFilterData(
+          dateRange: range,
+          text: range != null
+              ? '${range.start._format} - ${range.end._format}'
+              : 'None',
+        ),
+      );
+
+  @override
+  Stream<CapacityFilterData> get capacityFilterData =>
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:115]: @DDavidPrime can we move the implementation details to the extension file ?

File: examples/booking_app/lib/feature_hotel/search/ui_components/advanced_filter_page.dart [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import 'package:booking_app/feature_hotel/search/ui_components/filter_value_chooser.dart';
+import 'package:flutter/material.dart';
+import 'package:rflutter_alert/rflutter_alert.dart';
+
+class AdvancedFilterPage extends StatefulWidget {
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/ui_components/advanced_filter_page.dart:5]: @DDavidPrime can we rename the widget to something like HotelCapacityPage ?

File: examples/booking_app/lib/feature_hotel/search/views/search_page.dart [hunk 7]
```diff
@@ -170,4 +145,198 @@ class _SearchPageState extends State<SearchPage> with TickerProviderStateMixin {
       ),
     );
   }
+
+  Widget _buildFilters(BuildContext context) => Padding(
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/views/search_page.dart:149]: @DDavidPrime can we break down this huge method to smaller once. In that way we would improve the readability. :)

File: examples/booking_app/lib/feature_hotel/search/views/search_page.dart [hunk 8]
```diff
@@ -170,4 +145,198 @@ class _SearchPageState extends State<SearchPage> with TickerProviderStateMixin {
...
+                    ],
+                  );
+                },
+              ),
+            ),
+          ],
+        ),
+      );
+
+  Future<void> _presentYesNoDialog({
```
Comment (StanevPrime) [inline examples/booking_app/lib/feature_hotel/search/views/search_page.dart:298]: @DDavidPrime can we have this as a separate widget so we can use it from other places too ?

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart (hunk 1 above)
Comment (DDavidPrime) [reply examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:7]: `DateTimeRange` is part of flutter's `material` library, so it is required in this case.

File: examples/booking_app/lib/feature_hotel/search/views/search_page.dart (hunk 7 above)
Comment (DDavidPrime) [reply examples/booking_app/lib/feature_hotel/search/views/search_page.dart:151]: Improved by separating its contents into two separate parts: one for each filter.

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart (hunk 2 above)
Comment (DDavidPrime) [reply examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:50]: Not really. Updated!

File: examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart (hunk 3 above)
Comment (DDavidPrime) [reply examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart:52]: Updated as the one above it!

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**